src/test/java/streams/part2/exercise/Exercise1.java

```diff
@@ -0,0 +1,159 @@
+package streams.part2.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+public class Exercise1 {
+
+    @Test
+    public void calcTotalYearsSpentInEpam() {
+        List<Employee> employees = getEmployees();
+
+        // TODO реализация
+        Long hours = employees.stream()
+                .map(Employee::getJobHistory)
+                .flatMap(Collection::stream)
+                .filter(job -> job.getEmployer().equals("EPAM"))
+                .map(JobHistoryEntry::getDuration)
+                .reduce(0, (a, b) -> a + b)
+                .longValue();
+
+        assertEquals(19, hours.longValue());
+    }
+
+    @Test
+    public void findPersonsWithQaExperience() {
+        List<Employee> employees = getEmployees();
+
+        // TODO реализация
+        Set<Person> workedAsQa = employees.stream()
+                .filter(empl -> empl.getJobHistory().stream()
+                        .anyMatch(job -> job.getPosition().equals("QA")))
```

Comment:
In that exercise that is fine, but it's better to write like this: ```"QA".equals(...)``` - this will help you to avoid ```NullPointerException```.